compiler/dialects/include/byteir/Dialect/Ccl/IR/CclOps.td

```diff
@@ -217,8 +215,6 @@ def Ccl_SendOp : Ccl_Op<"send", [
 }
 
 def Ccl_RecvOp : Ccl_Op<"recv", [
-    SameOperandsAndResultElementType,
-    SameOperandsAndResultShape,
```

Comment:
Don't need to remove these when bufferize to lccl dialect?

Reply:
The dynamic case is considered here.

Reply:
If you remove this, you should add check for `ShapedType` in verify function.